Rx.NET/Documentation/Rx-Roadmap-2023.md

```diff
@@ -0,0 +1,170 @@
+
+We would dearly love to rip out all of the UWP-specific code in Rx, because it would simplify the build and fix a lot of problems. But since developers still sometimes find themselves with no option but to use UWP, Rx needs to continue to work on UWP. However, it's worth noting that these two things are not the same:
+
+1. supporting UWP
+2. providing UWP-specific targets in `System.Reactive`
+
+We believe that 1 continues to be necessary, but we would prefer to stop doing 2. We will provide full support for UWP but ideally, we would do this with all UWP-specific code separated out of the main solution. Unfortunately, a small amount of UWP-specific API surface area has crept into `System.Interactive`. (The `ThreadPoolScheduler` looks slightly different in the UWP targets, because it provides properties enabling you to set the [`WorkItemPriority`](https://learn.microsoft.com/en-us/uwp/api/windows.system.threading.workitempriority) and [`WorkItemOptions`](https://learn.microsoft.com/en-us/uwp/api/windows.system.threading.workitemoptions) it should use.) Our current view is that since this already represents a divergence between the API surface area of the UWP target from every other target, that it would be worth tolerating a breaking change that affects only UWP code in order to remove the UWP-specific target from `System.Reactive`. But regardless of how we achieve it, one way or another we need to continue to provide support for running on UWP for at least the next couple of versions.
+
+We have produced a prototype demonstrating what Rx might look like if the UWP parts were separated: https://github.com/idg10/reactive/pull/3. Since this removes the UWP target from `System.Reactive`, this includes a breaking change for UWP clients: they will find that the `ThreadPoolScheduler` now looks identical to the ones available to all other Rx clients; if they need the old functionality it is available through a new `WindowsThreadPoolScheduler` in the `System.Reactive.Windows.Threading` library, which is identical to the old UWP-specific `ThreadPoolScheduler` in everything but name.
+
+
+## Plans for upcoming release
+
+We need to decide whether we should try to solve all of the problems in one step, or whether we produce any interim releases that get us part way to a solution. Our current view is that we will not attempt to do everything at once. We would like to get automated tests running against .NET 6.0 and .NET 7.0 in place as soon as possible so that we can verify that these platforms are supported. (This necessarily entails fixing the problems that prevent Rx from being built using the latest tool versions.) And we would like to make preview builds of AsyncRx.NET available as quickly as possible.
```

Comment:
Do you think we should be explicit about making the async release an `alpha` release? We want to get it into people's hands, but there is much engineering (testing) work to do before it's in a non-preview state?

Reply:
I've replaced "preview" with `alpha` to clarify this.